src/services/formatting/formatting.ts

```diff
@@ -759,9 +759,15 @@ namespace ts.formatting {
                 if (child.kind === SyntaxKind.JsxText) {
                     const range: TextRange = { pos: child.getStart(), end: child.getEnd() };
                     if (range.pos !== range.end) { // don't indent zero-width jsx text
-                        const siblings = parent.getChildren(sourceFile);
-                        const currentIndex = findIndex(siblings, arg => arg.pos === child.pos);
-                        const previousNode = siblings[currentIndex - 1];
+                        let tempNode: Node;
+                        let previousNode: Node | undefined;
+                        forEachChild(parent, childNode => {
+                            if (childNode.pos === child.pos) {
+                                previousNode = tempNode;
+                                return true;
+                            }
+                            tempNode = childNode;
+                        });

```

Comment:
What is this change for?

Reply:
It fixes a bug for when the `JsxElement` contains multiple nodes. It used to be that only the first node would be formatted. 

In the scenario below only `some text` would be formatted before the fix:
```
const foo = <div>
             some text
                            {bar}
                    more text
</div>
```

Reply:
I guess I mean, I don't understand how the new code manages to do something different from the old code. They look like they should do the same thing, except that `forEachChild` doesn’t include non-node children (like punctuation tokens). But it seems like that doesn’t matter in your example.

Reply:
Got it. Before it used to search for the siblings of `JsxText` using the `getChildren` method of the parent. This will result in not being able to find the correct sibling as is will be looking for `JsxOpeningElement`, `JsxClosingElement` and in this example a `SyntaxList`. Non of them is really a sibling.

Instead, I have changed it to use `ForEachChild` which now the search will include the missing `JsxExpression` and `JsxText`, thus being able to correctly format the source code.

Just for visualization
With getChildren:
![image](https://user-images.githubusercontent.com/13305542/109567877-ef1adf00-7a9a-11eb-8e1e-d73e47eb5af9.png)

With forEachChild:
![image](https://user-images.githubusercontent.com/13305542/109567919-fe9a2800-7a9a-11eb-9cb5-b690082d68e4.png)

Reply:
Ah, I didn’t realize `getChildren()` was returning a SyntaxList. That makes sense. Just as a small readability optimization, I would write it like this:

```ts
if (isJsxText(child)) {
    const range: TextRange = { pos: child.getStart(), end: child.getEnd() };
    if (range.pos !== range.end) { // don't indent zero-width jsx text
        const previousNode = child.parent.children[child.parent.children.indexOf(child) - 1] || child.parent.openingElement;
```

You can simplify a lot because you know that `child` is `JsxText` here.